File: projects/Lifes/src/Lifes.Core/Models/MementoModel.cs
using System;

namespace Lifes.Core.Models;

public class MementoModel
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    
    // Hierarchical link: null means Topic Note, non-null means Supplemental Concept Note
    public int? ParentId { get; set; }
    
    // Sort order
    public int Order { get; set; }
    
    public DateTime CreatedDate { get; set; } = DateTime.Now;
    
    // Visual category/color
    public string Color { get; set; } = string.Empty;
    
    public string Description { get; set; } = string.Empty;
    
    public List<int> TagIds { get; set; } = new();
}

File: projects/Lifes/src/Lifes.Presentation.Electron/src/app/models/memento.model.ts
export interface Memento {
  id: number;
  title: string;
  parentId: number | null;
  startDate: string; // ISO string
  endDate: string;   // ISO string
  order: number;
  color?: string;
  tagIds: number[];
  isAchieved: boolean;
  children?: Memento[];
}
